core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -190,6 +190,7 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
       .orNull
     numExecutors = Option(numExecutors)
       .getOrElse(sparkProperties.get("spark.executor.instances").orNull)
+    queue = Option(queue).orElse(sparkProperties.get("spark.yarn.queue")).getOrElse("default")
```

Comment:
I would be in favor of `orNull` instead of "default". `SparkSubmitArguments` reflects the arguments we set, since we don't set it, I think it would be better to be `null`. You could see other configurations here like `spark.driver.memory`, `spark.deploy.mode`, etc. Most of them indeed have default values, but here still use `null`.